pkg/apis/core/validation/validation.go

```diff
@@ -5616,6 +5616,37 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Forbidden(specPath, "Pod running on node without support for resize"))
 	}
 
+	// Do not allow adding, removing, re-ordering, or renaming containers on resize.
+	var containerOrderModified bool
+	if len(newPod.Spec.Containers) != len(oldPod.Spec.Containers) {
+		allErrs = append(allErrs, field.Forbidden(specPath.Child("containers"), "containers may not be added or removed on resize"))
+		containerOrderModified = true
+	} else {
+		for i, ctr := range oldPod.Spec.Containers {
```

Comment:
nit: rename `ctr` to `oldCtr` for clairity. Same below.
```suggestion
		for i, oldCtr := range oldPod.Spec.Containers {
```

Reply:
done